.github/workflows/pre-commit.yml

```diff
@@ -10,15 +10,16 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - uses: actions/setup-python@v2
-      with:
-        python-version: '3.8'
-    - uses: actions/setup-node@v2
-      with:
-        node-version: 10.x
-    - name: Install Modules
+
+    - name: Run pre-commit
       run: |
-        cd client
-        npm ci
+        git fetch origin main ${{ github.event.pull_request.base.sha }}
+        bin/infra build
+        touch dev.env
+        bin/dev run pipenv run pip install pre-commit
+        bin/dev sudo pre-commit run --hook-stage push --from-ref ${{ github.event.pull_request.base.sha }} --to-ref ${{ github.sha }}
```

Comment:
Short cutting the sudo install and chown required for local dev here because we don't care that files written by pre-commit are owned by root inside the gh action runner.

Reply:
How does `sudo` work within gh action

Reply:
`Sudo` doesn't actually invoke super user do, it just runs the `docker-compose run` command without specifying which user to run as. This implicitly runs the commands in the container with super user permissions.

Reply:
I see, not sure if it's a misnomer. opinions on changing it to `elevated` or something on a future pr?

Reply:
Rather than a misnomer, it's a translation from what avg dev expects sudo to do and what docker requires for super user commands.